src/security/context.cpp

```diff
@@ -962,7 +722,7 @@ ngx_int_t Context::do_request_body_filter(ngx_http_request_t &request,
       auto *conf = static_cast<datadog_loc_conf_t *>(
           ngx_http_get_module_loc_conf(&request, ngx_http_datadog_module));
 
-      if (task_ctx.submit(conf->waf_pool)) {
+      if (std::move(task_ctx).submit(conf->waf_pool)) {
```

Comment:
I assume this is related to what I comment on above but perhaps it deserves a comment / explanation here.